fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java

```diff
@@ -523,6 +523,48 @@ public boolean isDecimalV3OrContainsDecimalV3() {
         return false;
     }
 
+    // This method defines the char type
+    // to support the schema-change behavior of length growth.
+    public boolean isSupportSchemaChangeForCharType(Type other) throws TypeException {
+        if ((this.getPrimitiveType() == PrimitiveType.VARCHAR && other.getPrimitiveType() == PrimitiveType.VARCHAR) || (
+                this.getPrimitiveType() == PrimitiveType.CHAR && other.getPrimitiveType() == PrimitiveType.VARCHAR) || (
+                this.getPrimitiveType() == PrimitiveType.CHAR && other.getPrimitiveType() == PrimitiveType.CHAR)) {
+            if (this.getLength() > other.getLength()) {
+                throw new TypeException("Cannot shorten string length");
+            } else {
+                return true;
```

Comment:
not return boolean value, just throw the ddl exception should be enough?

Reply:
done!